src/blockchain.js

```diff
@@ -91,8 +91,8 @@ Blockchain.doConnect = function(connectionList, opts, doneCb) {
       try {
         if (Blockchain.autoEnable) {
           await ethereum.enable();
-          this.blockchainConnector.setProvider(ethereum);
         }
+        this.blockchainConnector.setProvider(ethereum);
```

Comment:
So this change "kind of" makes sense to me, but I remember that this was by design no? `ethereum` should only set as provider when access is granted, otherwise it'll just stick to e.g. web3.

Chances are I'm missing something here.

Reply:
👌